x-pack/plugins/fleet/server/services/agents/status.ts

```diff
@@ -92,3 +94,62 @@ export async function getAgentStatusForAgentPolicy(
     events: 0,
   };
 }
+export async function getIncomingDataByAgentsId(
+  esClient: ElasticsearchClient,
+  agentsIds: string[]
+) {
+  try {
+    const searchResult = await esClient.search({
+      index: DATA_STREAM_INDEX_PATTERN,
```

Comment:
It is possible the current user do not have access to these datastreams. Should we check the user privileges before and throw an error if it's the case?

Reply:
I added a check for the user privileges, it should now throw an error if they're not present,